gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AllergyIntoleranceMapper.java

```diff
@@ -120,9 +120,13 @@ private void buildCode(AllergyIntolerance allergyIntolerance, RCMRMT030101UK04Co
         if (NON_DRUG_ALLERGY_CODE.equals(compoundCode)) {
             allergyIntolerance.setCode(codeableConceptFromCode);
 
-            if (allergyIntolerance.getCode() != null && !allergyIntolerance.getCode().hasCoding()) {
-                allergyIntolerance.getCode().setCoding(List.of(DegradedCodeableConcepts.DEGRADED_NON_DRUG_ALLERGY));
+            if (allergyIntolerance.getCode() == null) {
+                return;
             }
+
+            DegradedCodeableConcepts.addDegradedEntry(
```

Comment:
I got confused by this code to begin with, but I think the problem is the naming of this method `addDegradedEntry` is more along the lines of `addDegradedEntryIfRequired`?

Reply:
Yes might make more sense, I'll change this and the other similar named method